bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -673,24 +688,26 @@ public void run()
             data = lData;
             readingOffset = 0;
             writingOffset = data.length;
-            if (refCount > 1) {
+            if (--numberOfInMemBlockPermits < 0) {
+              throw new RuntimeException("Exceeded allowed memory block allocation by " + (-numberOfInMemBlockPermits));
+            }
+            if (refCount.get() > 1) {
               Block.this.notifyAll();
             }
           }
         }
-
       };
     }
-
-    synchronized void acquire(boolean wait)
+    
+    protected void acquire(boolean wait)
     {
-      if (refCount++ == 0 && uniqueIdentifier > 0 && storage != null) {
-        assert (data == null);
+      if (refCount.getAndIncrement() == 0 && storage != null && data == null) {
```

Comment:
No, it is not an error case anymore. Note that in release() we decrement the count and **schedule** write to disk. In case acquire() is called between call to release() and when storer is run, the data will not be set to null, while refCount can be 0.